modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2289 @@
+
+  params.addParam<std::vector<std::string>>(
+      "energy_inlet_function",
+      std::vector<std::string>(),
+      "Functions for fixed-value boundaries in the energy equation.");
+
+  MultiMooseEnum en_wall_types("fixed-temperature heatflux symmetry", "symmetry");
+  params.addParam<MultiMooseEnum>(
+      "energy_wall_types", en_wall_types, "Types for the wall boundaries for the energy equation.");
+
+  params.addParam<std::vector<FunctionName>>(
+      "energy_wall_function",
+      std::vector<FunctionName>(),
+      "Functions for Dirichlet/Neumann boundaries in the energy equation.");
+
+  params.addParam<std::vector<SubdomainName>>(
+      "ambient_convection_blocks",
+      std::vector<SubdomainName>(),
+      "The blocks where the ambient convection is present.");
+
+  params.addParam<std::vector<MooseFunctorName>>(
+      "ambient_convection_alpha",
+      std::vector<MooseFunctorName>(),
+      "The heat exchange coefficients for each block in 'ambient_convection_blocks'.");
+
+  params.addParam<std::vector<MooseFunctorName>>(
+      "ambient_temperature",
+      std::vector<MooseFunctorName>(),
+      "The ambient temperature for each block in 'ambient_convection_blocks'.");
+
+  params.addParam<CoupledName>(
+      "external_heat_source",
+      "The name of a functor which contains the external heat source for the energy equation.");
+  params.addParam<Real>(
+      "external_heat_source_coeff", 1.0, "Multiplier for the coupled heat source term.");
+
+  /**
+   * Parameters controlling the friction terms in case of porous medium simulations.
+   */
+
+  params.addParam<std::vector<SubdomainName>>(
+      "friction_blocks",
+      "The blocks where the friction factors are applied to emulate flow resistances.");
+
+  params.addParam<std::vector<std::vector<std::string>>>(
+      "friction_types", "The types of friction forces for every block in 'friction_blocks'.");
+
+  params.addParam<std::vector<std::vector<std::string>>>(
+      "friction_coeffs",
+      "The firction coefficients for every item in 'friction_types'. Note that if "
+      "'porous_medium_treatment' is enabled, the coefficients already contain a velocity "
+      "multiplier but they are not multiplied with density yet!");
+
+  params.addParamNamesToGroup("friction_blocks friction_types friction_coeffs",
+                              "Friction controls");
+
+  /**
+   * Parameters describing the handling of advected scalar fields
+   */
+
+  params.addParam<std::vector<NonlinearVariableName>>(
+      "passive_scalar_names",
+      std::vector<NonlinearVariableName>(),
+      "Vector containing the names of the advected scalar variables.");
+
+  params.addParam<std::vector<bool>>(
+      "create_scalar_variable",
+      std::vector<bool>(),
+      "Switch to indicate that the input nonlinear variables don't need to be "
+      "generated within this action.");
+
+  params.addParam<std::vector<Real>>("initial_scalar_variables",
```

Comment:
same, down for functors here. I initialize the MSFR with functions as I've had trouble with scaling with 0 (admittedly never tried another constant)

Reply:
I added functions, I'll leave functorICs for another PR.